cdk/apps/ica_credentials/cdk/secrets/infrastructure.py

```diff
@@ -290,31 +323,31 @@ def share_jwt_secret_with_github_actions_repo(
             return
 
         # Set role
-        gh_action_role = Role(
+        gh_action_role = iam.Role(
             self,
             role_name,
-            assumed_by=FederatedPrincipal(
-                f"arn:aws:iam::{account_id}:oidc-provider/token.actions.githubusercontent.com",
+            assumed_by=iam.FederatedPrincipal(
+                "arn:aws:iam::"
+                + Stack.of(self).account
+                + ":oidc-provider/token.actions.githubusercontent.com",
```

Comment:
This is better because we shouldn't need to pass in the account - the Stack already knows where it is being installed